completions/bash/multipass

```diff
@@ -21,25 +21,105 @@ _multipass_complete()
         local cmd="multipass list --format=csv --no-ipv4"
         [ -n "$state" ] && cmd="$cmd | \awk -F',' '\$2 == \"$state\"'"
 
-        local instances=$( \eval $cmd | \grep -Ev '(\+--|Name)' | \cut -d',' -f 1 )
+        local retval=$( \eval $cmd | \grep -Ev '(\+--|Name)' | \cut -d',' -f 1 )
+
+        for instance in $retval; do
+            instances="${instances} ${instance}"
+        done
+    }
+
+    _multipass_snapshots()
+    {
+        local instance="${cur%%.*}"
+        local cmd="multipass info --snapshot-overview "${instance}" --format=csv 2>/dev/null"
+        local retval=$( \eval $cmd | \grep -Ev '(\+--|Instance)' | \cut -d',' -f 2 )
+
+        for snapshot in $retval; do
+            snapshots="${snapshots} ${instance}.${snapshot}"
+        done
+    }
+
+    _multipass_instances_and_snapshots()
+    {
+        if [[ "${cur}" =~ '.' ]]; then
+            _multipass_snapshots
+            _multipass_filter_array "snapshots"
+        fi
+
+        _multipass_instances
+        _multipass_filter_array "instances"
+    }
+
+    _multipass_filter_array()
+    {
+        local backup=()
+
+        if [[ $1 == "instances" ]]; then
+            backup=$instances
+            instances=""
+        elif [[ $1 == "snapshots" ]]; then
+            backup=$snapshots
+            snapshots=""
+        fi
 
         local found
+        _get_comp_words_by_ref -n := -w WORDS -i CWORD cur prev
```

Comment:
I don't think this needs the colon and the equals. That incantation tells it not to consider those two characters as word splitters, for cases like instance:dir or key=val. Not really applicable here, I don't think.